src/framework/theme/components/autocomplete/autocomplete.directive.ts

```diff
@@ -0,0 +1,325 @@
+  protected getDisplayValue(value: string) {
+    const displayFn = this.autocomplete.handleDisplayFn;
+    return displayFn ? displayFn(value) : value;
+  }
+
+  protected getContainer() {
+    return this.overlayRef && this.isOpen && <ComponentRef<any>> {
+      location: {
+        nativeElement: this.overlayRef.overlayElement,
+      },
+    };
+  }
+
+  protected handleOptionClick(option: NbOptionComponent<T>) {
+
+    this.setHostInputValue(option.value);
+    this._onChange(option.value);
+
+    this.hostRef.nativeElement.focus();
+    this.autocomplete.emitSelected(option.value);
+    this.hide();
+  }
+
+  protected subscribeOnTriggers() {
+
+    this.triggerStrategy.show$
+      .pipe(filter(() => this.isClosed))
+      .subscribe(($event: Event) => {
+        this.show();
+      });
+
+    this.triggerStrategy.hide$
+      .pipe(filter(() => this.isOpen))
+      .subscribe(($event: Event) => {
+        this.hide();
+      });
+  }
+
+  protected createTriggerStrategy(): NbTriggerStrategy {
+    return this.triggerStrategyBuilder
+      .trigger(NbTrigger.FOCUS)
+      .host(this.hostRef.nativeElement)
+      .container(() => this.getContainer())
+      .build();
+  }
+
+  protected createKeyManager(): void {
+    this.keyManager = new NbActiveDescendantKeyManager<NbOptionComponent<T>>(this.autocomplete.options).withWrap();
+  }
+
+  show() {
+    if (this.isClosed) {
+      this.attachToOverlay();
+      this.setActiveItem();
+    }
+  }
+
+  hide() {
+    if (this.isOpen) {
+      this.overlayRef.detach();
+      this.setActiveItem();
+    }
+  }
+
+  protected setHostInputValue(value) {
+    this.hostRef.nativeElement.value = this.getDisplayValue(value);
+  }
+
+  protected createPositionStrategy(): NbAdjustableConnectedPositionStrategy {
+    return this.positionBuilder
+      .connectedTo(this.hostRef)
+      .position(NbPosition.BOTTOM)
+      .offset(0)
+      .adjustment(NbAdjustment.VERTICAL);
+  }
+
+  protected subscribeOnOverlayKeys(): void {
+    this.overlayRef.keydownEvents()
+      .pipe(
+        takeUntil(this.autocomplete.destroy$),
+      )
+      .subscribe((event: KeyboardEvent) => {
+        if (event.keyCode === ESCAPE && this.isOpen) {
+          this.hostRef.nativeElement.focus();
+          this.hide();
+
+        } else if (event.keyCode === ENTER) {
+          const activeItem = this.getActiveItem();
+          if (!activeItem) {
+            return;
+          }
+          this._onChange(activeItem.value);
+          this.setHostInputValue(activeItem.value);
+          this.autocomplete.emitSelected(activeItem.value);
+          this.hide();
+
+        } else {
+          this.keyManager.onKeydown(event);
+        }
+      });
+
+    this.keyManager.tabOut
+      .pipe(takeUntil(this.autocomplete.destroy$))
+      .subscribe(() => {
+        this.hide();
+      });
+  }
+
+  protected setActiveItem() {
+    this.keyManager.setActiveItem(this.autocomplete.activeFirst ? 0 : -1);
+    this.cd.detectChanges();
```

Comment:
Because dom for option isn't update (active class) without run change detection manually, autocomplete component with option list is sibling for host input.